opentelemetry-sdk/src/opentelemetry/sdk/trace/propagation/jaeger_format.py

```diff
@@ -0,0 +1,129 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+import urllib.parse
+
+import opentelemetry.trace as trace
+from opentelemetry import baggage
+from opentelemetry.context import Context, get_current
+from opentelemetry.trace.propagation.textmap import (
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+
+class JaegerFormat(TextMapPropagator):
+    """Propagator for the Jaeger format.
+
+    See: https://www.jaegertracing.io/docs/1.19/client-libraries/#propagation-format
+    """
+
+    TRACE_ID_KEY = "uber-trace-id"
+    BAGGAGE_PREFIX = "uberctx-"
+
+    def extract(
+        self,
+        getter: Getter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> Context:
+
+        if context is None:
+            context = get_current()
+        fields = _extract_first_element(
+            getter.get(carrier, self.TRACE_ID_KEY)
+        ).split(":")
+        if len(fields) == 4:
+            trace_id, span_id, _parent_id, flags = fields
+        else:
+            return trace.set_span_in_context(trace.INVALID_SPAN)
+        span = trace.DefaultSpan(
+            trace.SpanContext(
+                trace_id=int(trace_id, 16),
+                span_id=int(span_id, 16),
```

Comment:
Should this check for a 0 value? `Value of 0 is not valid`

by convention (and this may change), an invalid span / trace id are represented by the integer value 0.

Reply:
that's a good point, I'll add a check for both trace and span id.